core/src/main/java/org/elasticsearch/search/aggregations/AggregatorParsers.java

```diff
@@ -60,10 +60,12 @@ public AggregatorParsers(Set<Aggregator.Parser> aggParsers, Set<PipelineAggregat
         Map<String, Aggregator.Parser> aggParsersBuilder = new HashMap<>(aggParsers.size());
         for (Aggregator.Parser parser : aggParsers) {
             aggParsersBuilder.put(parser.type(), parser);
-            AggregatorFactory factoryPrototype = parser.getFactoryPrototype();
+            AggregatorFactory[] factoryPrototypes = parser.getFactoryPrototypes();
             // NORELEASE remove this check when agg refactoring complete
-            if (factoryPrototype != null) {
-                namedWriteableRegistry.registerPrototype(AggregatorFactory.class, factoryPrototype);
+            if (factoryPrototypes != null) {
```

Comment:
I absolutely agree. This null check is only here while there are Aggregations which are not refactored. Once the aggregation refactoring is complete this check will be removed and the method will need to return a non-null value (see NORELEASE comment above). I could go through and make all non refactored aggregation return an empty array for this method but I wanted to leave them returning null so that if this check is removed before all aggregations have been refactored we will be alerted to it as the node will throw a NPE on start up.